src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -4,61 +4,317 @@
     {
-        const CoreWindow window = CoreWindow::GetForCurrentThread();
-        const auto rAltState = window.GetKeyState(VirtualKey::RightMenu);
-        const auto lAltState = window.GetKeyState(VirtualKey::LeftMenu);
-        const bool altPressed = WI_IsFlagSet(lAltState, CoreVirtualKeyStates::Down) ||
-                                WI_IsFlagSet(rAltState, CoreVirtualKeyStates::Down);
+        if (args.OldKeys().Modifiers() != args.NewKeys().Modifiers() || args.OldKeys().Vkey() != args.NewKeys().Vkey())
+        {
+            // We're actually changing the key chord
+            const auto senderVMImpl{ get_self<KeyBindingViewModel>(senderVM) };
+            const auto& conflictingCmd{ _State.Settings().ActionMap().GetActionByKeyChord(args.NewKeys()) };
+            if (conflictingCmd)
+            {
+                // We're about to overwrite another key chord.
+                // Display a confirmation dialog.
+                TextBlock errorMessageTB{};
+                errorMessageTB.Text(RS_(L"Actions_RenameConflictConfirmationMessage"));
+
+                const auto conflictingCmdName{ conflictingCmd.Name() };
+                TextBlock conflictingCommandNameTB{};
+                conflictingCommandNameTB.Text(fmt::format(L"\"{}\"", conflictingCmdName.empty() ? RS_(L"Actions_UnnamedCommandName") : conflictingCmdName));
+                conflictingCommandNameTB.FontStyle(Windows::UI::Text::FontStyle::Italic);
+
+                TextBlock confirmationQuestionTB{};
+                confirmationQuestionTB.Text(RS_(L"Actions_RenameConflictConfirmationQuestion"));
+
+                Button acceptBTN{};
+                acceptBTN.Content(box_value(RS_(L"Actions_RenameConflictConfirmationAcceptButton")));
+                acceptBTN.Click([=](auto&, auto&) {
+                    // remove conflicting key binding from list view
+                    const auto containerIndex{ _GetContainerIndexByKeyChord(args.NewKeys()) };
+                    _KeyBindingList.RemoveAt(*containerIndex);
+
+                    // remove flyout
+                    senderVM.AcceptChangesFlyout().Hide();
+                    senderVM.AcceptChangesFlyout(nullptr);
 
-        const auto target = altPressed ? SettingsTarget::DefaultsFile : SettingsTarget::SettingsFile;
+                    // update settings model and view model
+                    _State.Settings().ActionMap().RebindKeys(args.OldKeys(), args.NewKeys());
+                    senderVMImpl->Keys(args.NewKeys());
+                    senderVM.ToggleEditMode();
+                });
 
-        _State.RequestOpenJson(target);
+                StackPanel flyoutStack{};
+                flyoutStack.Children().Append(errorMessageTB);
+                flyoutStack.Children().Append(conflictingCommandNameTB);
+                flyoutStack.Children().Append(confirmationQuestionTB);
+                flyoutStack.Children().Append(acceptBTN);
+
+                Flyout acceptChangesFlyout{};

```

Comment:
_I wish!_ As mentioned in the comments below:
1. The data template prevents us from getting a control by name. So we have no way of targeting the button and attaching a pre-saved flyout.
2. The flyout needs to appear _sometimes_. So we can't just build it in XAML, because then we don't have control over when it appears or not.

Reply:
Ah I see, dang that's pretty annoying 😢

Reply:
I'm unsure if this will work, but you _might_ be able to give the button a flyout without having it automagically open the flyout on every button press by putting the flyout in `FlyoutBase.AttachedFlyout`. It's an attached property so this particular property can be set on just about any `FrameworkElement`, but IIRC it doesn't behave the same as if you set the `Flyout` property on Button.

https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.primitives.flyoutbase?view=winrt-19041#xaml-attached-properties - should tell you more about how to use this particular attached property

This is more an experimental idea though, not something I'd block over 😄 

e.g.
```
<Button>
    <FlyoutBase.AttachedFlyout>
        <Flyout>
        </Flyout>
    </FlyoutBase.AttachedFlyout>
</Button>
```